Lib/multiprocessing/connection.py

```diff
@@ -341,20 +356,44 @@ def _recv_bytes(self, maxsize=None):
             raise RuntimeError("shouldn't get here; expected KeyboardInterrupt")
 
         def _poll(self, timeout):
-            if (self._got_empty_message or
-                        _winapi.PeekNamedPipe(self._handle)[0] != 0):
-                return True
+            try:
+                if (self._got_empty_message or
+                            _winapi.PeekNamedPipe(self._handle)[0] != 0):
+                    return True
+            except OSError as err:
+                if err.errno == errno.EBADF:
```

Comment:
We should not get EBADF. We should just call functions on closed handle. If we land in this case, we failed badly in before, no?

Reply:
How can we know that it was closed in other thread without using locks?

You can get EBADF in the following scenario:

* Thread A reads `self._handle` and puts in on the stack.
* Thread B closes it.
* Thread A calls function with an already closed file descriptor on the stack.
* That function fails with EBADF errno.

It is the bast case. The worst case is:

* Thread A reads `self._handle` and puts in on the stack.
* Thread B closes it.
* Thread B or C opens a new file descriptor which reuses the same integer value.
* Thread A calls function with a file descriptor  which now refers to unrelated file or socket.
* Unpredictable consequences.